GiniVision/Classes/ReviewViewController.swift

```diff
@@ -162,6 +200,22 @@ public typealias ReviewErrorBlock = (_ error: ReviewError) -> ()
         DispatchQueue.main.async {
             (self.topView as? NoticeView)?.show()
         }
+        
+        if let currentDocument = currentDocument {
+            // Validate document before review
+            do {
+                try currentDocument.validate()
+            } catch let error as DocumentValidationError {
+                failureBlock!(error)
+            } catch _ {
+                failureBlock!(DocumentValidationError.unknown)
+            }
+            
+            // If the document is a PDF, go to Analysis screen
+            if currentDocument.type == .PDF {
```

Comment:
This decision should be made outside of the `ReviewViewController`, either when launching the library (open with) or when a file was picked in the `CameraViewController`.

Reply:
Related to previous comments.